text/trace/0220-messaging-semantic-conventions-span-structure.md

```diff
@@ -0,0 +1,656 @@
+
+#### Settlement
+
+Messages can be settled in a variety of different ways:
+
+* The intermediary settles the messages as it is sent to the consumer. No
+  settlement operations happen on the consumer.
+* The consumer settles a message without awaiting an acknowledgment from the
+  intermediary.
+* The consumer settles a message and awaits an acknowledgment from the
+  intermediary. This involves a round-trip exchange between the consumer and
+  the intermediary.
+
+Settlement operations on the consumer can either be triggered manually by the
+user, or can be triggered automatically by messaging SDKs based on return
+values of callbacks.
+
+A "Settle" span should be created for every settlement operation that happens
+on the consumer (at-least-once and exactly-once). SDKs will, in some cases,
+auto-settle messages in push-scenarios when messages are delivered via
+callbacks.
+
+"Settle" spans should link to creation context of the messages that are
+settled, when possible.
+
+No settlement span should be created for settlement scenarios that do not
+involve any settlement operations on the consumer side.
+
+## Proposed changes and additions to the messaging semantic conventions
+
+### Operation name
+
+The following operations related to messages are covered by these semantic
+conventions:
+
+| Operation name | Description |
+|----------------|-------------|
+| `publish`      | One or more messages are provided for publishing to an intermediary. |
+| `create`       | A message is created. |
+| `receive`      | One or more messages are requested by a consumer. |
+| `deliver`      | One or more messages are passed to a consumer. |
+| `settle`       | One or more messages are settled. |
+
+For further details about each of those operations refer to the [section about trace structure](#trace-structure).
+
+### Span kind
+
+[Span kinds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind)
+SHOULD be set according to the following table, based on the operation a span describes.
+
+| Operation name | Span kind|
+|----------------|-------------|
+| `publish`      | `PRODUCER`, if no `create` spans are present. |
+| `create`       | `PRODUCER` |
+| `receive`      | `CONSUMER` |
+| `deliver`      | `CONSUMER` |
+
+The kind of `settle` spans should be set according to the [generic specification about span kinds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind),
+e. g. it should be set to `CLIENT` if the `settle` spans models a synchronous call
+to the intermediary.
+
+Setting span kinds according to this table ensures that span links between
+consumers and producers always exist between a `PRODUCER` span on the producer
+side and a `CONSUMER` span on the consumer side. This allows analysis tools to
+interpret linked traces without the need for additional semantic hints.
+
+### Span relationships
```

Comment:
This seems like a rehash of the "Trace structure" section? Please unify, or add a note about the relationship of this section to "Trace structure".

Reply:
Then I propose:

```suggestion
### Span relationships

This section defines semantic conventions for the high-level structure outlined in [Trace structure](#trace-structure) more closely. In case of any contradictions, this section here overrides anything in the former section.
```

Reply:
I added this paragraph at the beginning of the "Proposed changes" section, I hope this makes things clearer:
 
 > This section contains a concise and normative definition of what was outlined in the [Trace structure](#trace-structure) section. The following subsections are supposed to be merged into the semantic conventions, whereas the detailed description in the [Trace structure](#trace-structure) section in this OTEP will serve as explanation and future reference.